### ISSUE TITLE
Please add a table template so can be easily added to the drawings

### ISSUE THREAD

**Issue by masoudshab** [excalidraw/excalidraw#4847]

**Comment by denisgolius** [comment]
Is there any information on adding this feature?

**Comment by ad1992** [comment]
do you mean adding it to library?

**Comment by denisgolius** [comment]
> do you mean adding it to library?

Yep, I've spend a few days to find third-party library for doing this without success. Also I can't find how to add curly bracket for text like in screenshot:

<img width="879" alt="vm-data-model" src="https://user-images.githubusercontent.com/5650611/166639878-de3466ec-686a-4019-be60-be00f121b6bb.png">

**Comment by denisgolius** [comment]
Even on excalidraw.com I didn't find any solution... Any thoughts?

**Comment by masoudshab** [comment]
> do you mean adding it to library?

Yes, having a table in the Excalidraw Library is very necessary. 
![image](https://user-images.githubusercontent.com/36493321/166689556-dc17429c-1ad1-4110-ad5f-9021354e7cf4.png)

A table that is flexible (writing in the cells expands rows and columns sizes), customizable (can be easily customized), and extendable (additional rows and columns can be easily added). A very good example is "Tables" in Whimsical ([whimsical.com](https://whimsical.com/login))
![image](https://user-images.githubusercontent.com/36493321/166690378-7c7324d2-fa91-472e-93ba-180b57555430.png)

**Comment by AuboIoT** [comment]
Yeah, I agree. I really like Excalidraw, but it's necessary a table.

**Comment by 2187Nick** [comment]
Table please

**Comment by joscul** [comment]
This would be a super awesome feature.

**Comment by gideondev** [comment]
Can't wait to have this feature!

**Comment by ryanwwest** [comment]
Me too.

**Comment by yanbussieres** [comment]
+1

**Comment by hector-sab** [comment]
+1

Currently I have rely on Lucid Charts for lots of things due to this missing feature.

**Comment by fcanmekikoglu** [comment]
+1

**Comment by dwelle** [comment]
Just to clarify our stance, we will definitely be adding this at some point, but it's not an easy task, and many things need to specced up.

**Comment by Jaitoh** [comment]
+1

**Comment by sean-scott-lr** [comment]
+1

**Comment by aaronjoseph** [comment]
+1

**Comment by sentinelleader** [comment]
+1

**Comment by butera-simone** [comment]
+1

**Comment by vincedprime** [comment]
+1

**Comment by tombiggs-nz** [comment]
+1

**Comment by chetnachaudhari** [comment]
+1

**Comment by MarcosRibeiroJesus** [comment]
Please 🙏😿

**Comment by sidPhoenix17** [comment]
Please 🙏🏽

**Comment by christianangelopoulos** [comment]
Please add a table!!! I want to use this for presentations and tables are a must! 

-an excalidraw plus subscriber :)

**Comment by YuvalItzchakov** [comment]
+1

**Comment by AndreiTuta** [comment]
+1. Would really love this feature

**Comment by FrankJingHao** [comment]
a table is really a must.
With table support, I would use nothing but Excalidraw, all in one.

**Comment by kaanyaltiunity** [comment]
Table support would take excalidraw to the next level

**Comment by hmu-duc-anh** [comment]
+1. Thank you for your consideration!

**Comment by valuereality** [comment]
+1

**Comment by art3xias23** [comment]
+1

**Comment by neelabalan-corestack** [comment]
+1

**Comment by Rikiub** [comment]
+1

**Comment by wade-tait** [comment]
+1

**Comment by lasmithusa** [comment]
+1. This would be great for class diagrams and ERDs / data modeling.

**Comment by shervinrad100** [comment]
+1. Defo needed

**Comment by lajbel** [comment]
+1

**Comment by akirosingh** [comment]
+1

**Comment by Michaelfosse** [comment]
+1

**Comment by DanielJGeiger** [comment]
I have an idea how to implement tables using a subtype of `ExcalidrawFrameElement` and tying child elements to cells, with a custom action for in-cell distribution. My approach would allow configuring the number of rows/columns and their widths/heights, without limiting cell contents to text.

(Due to the technical specifics, such an implementation would not be available on https://excalidraw.com or Excalidraw Plus, until significant portions of #6037 were merged.  Until then, such an implementation would be available on https://math.preview.excalidraw.com.)

**Comment by mahmoudmagdy-7** [comment]
+1

**Comment by hakkm** [comment]
+1

**Comment by petrovxyz** [comment]
+1

**Comment by Kernel-rb** [comment]
+1

**Comment by rxd90** [comment]
+1

**Comment by huayemao** [comment]
+1

**Comment by huayemao** [comment]
> I have an idea how to implement tables using a subtype of `ExcalidrawFrameElement` and tying child elements to cells, with a custom action for in-cell distribution. My approach would allow configuring the number of rows/columns and their widths/heights, without limiting cell contents to text.
> 
> (Due to the technical specifics, such an implementation would not be available on https://excalidraw.com or Excalidraw Plus, until significant portions of #6037 were merged. Until then, such an implementation would be available on https://math.preview.excalidraw.com.)

is this available now? how to use?

**Comment by nials-relevant** [comment]
+1

**Comment by praveensrao** [comment]
+1

**Comment by qaynam** [comment]
+1

**Comment by nafisk** [comment]
+1

**Comment by koonopek** [comment]
+1

**Comment by montanier** [comment]
+1

**Comment by Dasanchez1992** [comment]
+1

**Comment by runfourestrun** [comment]
would love this.

**Comment by andreisfedotov** [comment]
Please:)

**Comment by nareshbulusu** [comment]
+1

**Comment by lucifercr07** [comment]
+1

**Comment by slyjain** [comment]
+1

**Comment by gabchx** [comment]
+1

**Comment by liormb** [comment]
Please prioritize table support

**Comment by dwelle** [comment]
Keeping this in mind as we navigate through other priorities!

**Comment by notAdev96** [comment]
Bumping this! As a Business Analyst, almost everything I do involves tables. Unfortunately I have had to stop using Excalidraw and shift to Microsoft Whiteboard (they integrate Loop tables), but I would really, really rather not. Please add tables to Excalidraw, my favorite app!

**Comment by luisnquin-invio** [comment]
Please!

**Comment by ad1992** [comment]
Hi I will start working on this soon, hopefully in July

**Comment by oubaydos** [comment]
+1

**Comment by nive-copia** [comment]
+1

**Comment by tanlda** [comment]
+1

**Comment by MehakFatima24** [comment]
+1

**Comment by VadimSchmitz** [comment]
+1

**Comment by itsngh** [comment]
+1 it'd be lovely to have tables in Excalidraw :D

**Comment by martincekada** [comment]
+1

**Comment by priyanshs** [comment]
+1

**Comment by nifaaz** [comment]
+1

**Comment by LeonardoRick** [comment]
+1

**Comment by zmxncbv123098** [comment]
+1

**Comment by Wael-Yassir** [comment]
+1

**Comment by eddyzags** [comment]
+1

**Comment by mlegait** [comment]
+1

**Comment by VaibhaveS** [comment]
+1

**Comment by ankitarya1019** [comment]
+1

**Comment by dib-a** [comment]
+1

**Comment by RiznykLeo** [comment]
+1

**Comment by jasmeet9323** [comment]
+1

**Comment by jazcona** [comment]
+1

**Comment by akshar001** [comment]
+1

**Comment by thiagomini** [comment]
In the meantime, for those looking for an alternative, I highly suggest writing tables in [markdown](https://www.markdownguide.org/).

This [website](https://www.tablesgenerator.com/markdown_tables#) is especially useful for generating the markdown code from a UI. Here's an example:

```md
# Person Table

| ID  | Name         | Age | Email                 | City        |
|-----|--------------|-----|-----------------------|-------------|
| 1   | John Doe     | 28  | john.doe@example.com  | New York    |
| 2   | Jane Smith   | 34  | jane.smith@example.com| Los Angeles |
| 3   | Alice Johnson| 22  | alice.j@example.com   | Chicago     |
| 4   | Bob Brown    | 45  | bob.brown@example.com | Houston     |
```

Then, you can wrap it in a rectangle to make it more familiar with excalidraw drawings:

![Image](https://github.com/user-attachments/assets/f2014037-e8b0-446d-b889-a3213b3f6c12)

The advantage is that it's way easier to add or remove elements from a markdown table, and many tool can easily parse and recognize it.

**Comment by LanceXuDev** [comment]
+1

**Comment by CengAmat** [comment]
+1

**Comment by igoriamushev** [comment]
+1

**Comment by cyrootman** [comment]
+1000000 :)

**Comment by obsidian-shades** [comment]
+1 😠

**Comment by FatahChan** [comment]
+1

**Comment by michaelfidelis** [comment]
+1

**Comment by mdjamilkashemporosh** [comment]
```js
while (!featureAdded) {
    counter += 1;
}
```

**Comment by michaelgutierrez0** [comment]
+1

**Comment by Nesqyk** [comment]
+1

**Comment by padiil** [comment]
+1

**Comment by sheidaev** [comment]
+1

**Comment by RaulBecerraB** [comment]
+1

**Comment by Jodekq** [comment]
+1

**Comment by alechter** [comment]
+1

**Comment by kevinwong1983** [comment]
+1

**Comment by tren03** [comment]
+1

**Comment by weber8thomas** [comment]
+1

**Comment by luckyfatpig** [comment]
hoping this feature

**Comment by waLLxAck** [comment]
+1

**Comment by alexdx0** [comment]
+1

**Comment by Lyroki17** [comment]
+1

**Comment by nicored** [comment]
+1

**Comment by neanda** [comment]
why can't this not be done?  it's such an obvious feature.  I was about to transfer my team away from Miro to this, but the lack of tables is too weird.  I'll check back in another couple of few years, but AI probably would've let me build my own one by then.

i hope you sort this out before it's too late, u have a good product.

all the libraries are very cool, but, wtf tables?

**Comment by martin-hirsch** [comment]
+1

**Comment by FieryWall** [comment]
+1

**Comment by jvargaski** [comment]
+1

**Comment by y2kshukla** [comment]
+1

**Comment by i-mozz** [comment]
+1

**Comment by johnlarkin1** [comment]
yikes this anthropic demo is about to blow this issue up....

**Comment by amrrs** [comment]
I thought a PR from claude code would be there already!!

**Comment by adekbadek** [comment]
Right? I was expecting to see the PR merged as part of the talk...

**Comment by Austin1serb** [comment]
Please allow copy and pasting of the table markdown as well, 
for example
```
 # | Question | Purpose
-- | -- | --
1 | What single metric = victory in 12 months? (profit, traffic, community uploads, press hits?) | Focuses the roadmap; kills vanity work.
2 | What minimum $/month makes the project self-funded? | Sets pricing + volume targets.
3 | What is our acceptable burn (time + cash) before first dollar in? | Prevents scope creep.
4 | Which monetization ladders will we refuse? (ads, paywall, data resale) | Locks ethical boundary conditions.

```
into:

![Image](https://github.com/user-attachments/assets/917dd5e2-cbd1-4a04-9c6a-f2daca474eac)

**Comment by myozawlatt** [comment]
+1 plz

**Comment by mrakauza** [comment]
+1

**Comment by q-hung** [comment]
👍

**Comment by maxcorbeau** [comment]
+1 hey @dwelle: first, absolutely 💓 excalidraw so thank you for it! Since it's been almost 3 years since you wrote ["we will definitely be adding this at some point"](https://github.com/excalidraw/excalidraw/issues/4847#issuecomment-1341648897) and some of us (me included) might be big frontend noobs, **maybe you could clarify why this feature is so tricky to implement and whether we will ever see it?** That would help set expectations.

**My use-case**: discussing data engineering architectures, want to have a recap table inside the drawing to compare pros/cons of each option

**Comment by chloe-codes1** [comment]
Also still waiting for this. Would love to see it added!

**Comment by MGGMStolte** [comment]
+1

**Comment by DimIsaev** [comment]
+1

**Comment by IceCoffee1024** [comment]
+1

**Comment by dcs-mateus-strappazzon** [comment]
+1

**Comment by yunatseng** [comment]
+1

**Comment by pagueru** [comment]
+1

**Comment by Oulan-Mansha** [comment]
[excalidrawTableGenerator.zip](https://github.com/user-attachments/files/21535120/excalidrawTableGenerator.zip)

Extract it, then go the extracted folder and double click on "index.html" file, then web page will be open in the browser. After that add your desired dimension and click on the generate and copy folder. Then go the excalidraw and paste it in excalidraw.
The table will be drawn in excalidraw

**Comment by raideno** [comment]
> [excalidrawTableGenerator.zip](https://github.com/user-attachments/files/21535120/excalidrawTableGenerator.zip)
> 
> Extract it, then go the extracted folder and double click on "index.html" file, then web page will be open in the browser. After that add your desired dimension and click on the generate and copy folder. Then go the excalidraw and paste it in excalidraw. The table will be drawn in excalidraw

I’ve embedded the tool into a website for easier use: [https://raideno.github.io/excalidraw-tools/](https://raideno.github.io/excalidraw-tools/).  

Everything is available in the repository [https://github.com/raideno/excalidraw-tools](https://github.com/raideno/excalidraw-tools), so you can contribute by adding new tools or improving the existing ones.  

I’ll probably add more features to the table generator later on (preview, more controls over the width of each column / row, more complex table layouts, etc).

Any feedback or improvement suggestions is welcome!

**Comment by matteraclement** [comment]
+1

**Comment by oacthiago** [comment]
@raideno, thank you very much for this. It's perfect.
Came here just to say how much I appreciate it.

**Comment by aumirza** [comment]
Hey folks,

I’ve built a utility app that solves this exact problem—Excalidraw’s lack of native table support.
I built (vibe coded) a utility app: [Excalidraw Table Maker](https://excalidraw-table-maker.vercel.app/) 

**What it does:**
- Converts HTML, Markdown, and CSV tables into Excalidraw-compatible diagrams.
- AI-powered table generation: Just give it a topic, and it creates a relevant table on the fly—perfect for brainstorming or quick documentation.


This tool aims to fill the gap until native table support is available. I would love feedback or ideas on how it could be improved or integrated more closely with Excalidraw!

**Comment by bjmc** [comment]
@aumirza this is really useful! Is the source available somewhere? Could the table elements start out "grouped" together when you paste them? And could it [support markdown syntax for text alignment?](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables#formatting-content-within-your-table) i.e. `:---` `---:` `:---:`

**Comment by sumitdotgh** [comment]
+1

**Comment by taqi26** [comment]
+1

**Comment by amitbha** [comment]
Pain points.

**Comment by bogomazov-std** [comment]
+1

**Comment by jannesblobel** [comment]
+1
